Assignment4/ArturoAS-y160324/Task07.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# **Task 07: Querying RDF(s)**
+
+# In[1]:
+
+
+get_ipython().system('pip install rdflib ')
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+
+# Leemos el fichero RDF de la forma que lo hemos venido haciendo
+
+# In[3]:
+
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+
+# **TASK 7.1: List all subclasses of "Person" with RDFLib and SPARQL**
+
+# In[5]:
+
+
+from rdflib.plugins.sparql import prepareQuery
+
+
+print("Parte de RDFLib:")
+ns = Namespace("http://somewhere#")
+for s,p,o in g.triples((None, RDFS.subClassOf, ns.Person)):
+  print(s,p,o)
+print() #space
+print("SPARQL:")
+query = prepareQuery('''
+  SELECT DISTINCT ?x
+  WHERE{
+          ?x (rdfs:subClassOf/rdfs:subClassOf*) ns:Person
```

Comment:
primer subClassOf no es necesario, porque `*` indica 0 o mas ocurrencias.

Reply:
> Assignment 4 ok, but you are adding changes to the notebooks. We said in class that no changes to notebooks would be accepted.

I dont know what you mean. How can i fix this? i just did the assignment on the notebooks and then converted them into python files since the specification said you wouldnt accept notebooks.

Reply:
You are deleting many of the assignments of your colleagues in this PR. I suggest you close it, you copy your changes into a separate folder, you do a fetch upstream, reclone the repo and then open a new PR with your changes only.

Reply:
FINALLY I could solve this git was giving me a lot of problems, had to do a reset. -----> new PR ----> #275